app/api/services/evidence.py

```diff
@@ -83,6 +84,112 @@ def get_previous_submitted_evidence_for_supplier_and_domain(self, evidence_id, d
         evidence = query.first()
         return evidence
 
+    def get_data(self, evidence_id):
+        domain_criteria_id = (
+            db.session.query(
+                func.json_array_elements_text(Evidence.data['criteria']).label('dc_id')
+            )
+            .filter(Evidence.id == evidence_id)
+            .subquery()
+        )
+
+        category_name = (
+            db.session.query(
+                Domain.name.label('category')
+            )
+            .filter(Evidence.id == evidence_id)
+            .join(Evidence, Evidence.domain_id == Domain.id)
```

Comment:
I think this reads better if we move the join before the filter.
```suggestion
            .join(Evidence, Evidence.domain_id == Domain.id)
            .filter(Evidence.id == evidence_id)
```